pkg/deploy/strategy/rolling/rolling.go

```diff
@@ -202,6 +210,24 @@ func (s *RollingDeploymentStrategy) Deploy(from *kapi.ReplicationController, to
 	return nil
 }
 
+// retry runs fn up to maxTries times until no error is returned, sleeping for
+// period between each try.
+func retry(maxTries int, period time.Duration, desc string, fn func() error) error {
```

Comment:
I think you want to use [wait.Poll](https://github.com/kubernetes/kubernetes/blob/0902f80f8b94fd23c66398118fcdf9e845c165b1/pkg/util/wait/wait.go#L49) which supports timeouts instead of maxTries